### ISSUE TITLE
Fails when azure-native schema in json locally and converting a specific token

### ISSUE THREAD

**Issue by wongstein** [pulumi/pulumi-converter-constructor-syntax#1]
### What happened?

Command:

pulumi convert --from constructor-syntax --language python --out auto-gen -- schema.json azure-native:network/frontDoor:FrontDoor

From what I can tell, pulumi package get-schema for aws-native has resources named in this format:
 "azure-native:network/v20210601:FrontDoorEnabledState": {}

Not sure how many other providers are like this.  This is true for azure-native latest and azure-native@<VERSION>

Tested: 
pulumi convert --from constructor-syntax --language python --out container-azure-python/gen-
code -- container-azure-python/schema.json azure-native:network/v20210601:FrontDoor

And still recieved same error, so it's probably both the token name being a problem AND syntax in the json.

### Example

Ask me for the json file, it's too big to upload on github

### Output of `pulumi about`

Here's the error:

Converting from constructor-syntax...
Converting to python...
================================================================================
The Pulumi CLI encountered a code generation error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v3.105.0
error: main.pp:385,6-10: Missing attribute separator; Expected a newline or comma to mark the beginning of the next attribute.
error: could not generate output program

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by wongstein** [comment]
also don't know if this helps, but in python json.load(FILE) works fine.

**Comment by Zaid-Ajaj** [comment]
Fixed as of v0.3.0 🚀 the problem was incorrect handling of union objects. 

The token provided in the command `azure-native:network/frontDoor:FrontDoor` doesn't actually exist as part of the resources, so the converter tried to generate examples for everything and failed when it encountered a union of objects.

Now it will error quickly if it doesn't find a resource or function with the specified token. The token you wanted was this:
```
azure-native:network:FrontDoor
```
You can uninstall the converter plugin using `pulumi plugin rm converter constructor-syntax` then running the following command again to get latest converter with the fix:
```
pulumi convert --from constructor-syntax --language python --out auto-gen -- schema.json azure-native:network:FrontDoor
```
for me, this generates the following code with the new v0.3.0
```python
import pulumi
import pulumi_azure_native as azure_native

example = azure_native.network.FrontDoor("example",
    resource_group_name="string",
    backend_pools=[azure_native.network.BackendPoolArgs(
        backends=[azure_native.network.BackendArgs(
            address="string",
            backend_host_header="string",
            enabled_state="string",
            http_port=0,
            https_port=0,
            priority=0,
            private_link_alias="string",
            private_link_approval_message="string",
            private_link_location="string",
            private_link_resource_id="string",
            weight=0,
        )],
        health_probe_settings=azure_native.network.SubResourceArgs(
            id="string",
        ),
        id="string",
        load_balancing_settings=azure_native.network.SubResourceArgs(
            id="string",
        ),
        name="string",
    )],
    backend_pools_settings=azure_native.network.BackendPoolsSettingsArgs(
        enforce_certificate_name_check="string",
        send_recv_timeout_seconds=0,
    ),
    enabled_state="string",
    friendly_name="string",
    front_door_name="string",
    frontend_endpoints=[azure_native.network.FrontendEndpointArgs(
        host_name="string",
        id="string",
        name="string",
        session_affinity_enabled_state="string",
        session_affinity_ttl_seconds=0,
        web_application_firewall_policy_link=azure_native.network.FrontendEndpointUpdateParametersWebApplicationFirewallPolicyLinkArgs(
            id="string",
        ),
    )],
    health_probe_settings=[azure_native.network.HealthProbeSettingsModelArgs(
        enabled_state="string",
        health_probe_method="string",
        id="string",
        interval_in_seconds=0,
        name="string",
        path="string",
        protocol="string",
    )],
    load_balancing_settings=[azure_native.network.LoadBalancingSettingsModelArgs(
        additional_latency_milliseconds=0,
        id="string",
        name="string",
        sample_size=0,
        successful_samples_required=0,
    )],
    location="string",
    routing_rules=[azure_native.network.RoutingRuleArgs(
        accepted_protocols=["string"],
        enabled_state="string",
        frontend_endpoints=[azure_native.network.SubResourceArgs(
            id="string",
        )],
        id="string",
        name="string",
        patterns_to_match=["string"],
        route_configuration=azure_native.network.ForwardingConfigurationArgs(
            odata_type="#Microsoft.Azure.FrontDoor.Models.FrontdoorForwardingConfiguration",
            backend_pool=azure_native.network.SubResourceArgs(
                id="string",
            ),
            cache_configuration=azure_native.network.CacheConfigurationArgs(
                cache_duration="string",
                dynamic_compression="string",
                query_parameter_strip_directive="string",
                query_parameters="string",
            ),
            custom_forwarding_path="string",
            forwarding_protocol="string",
        ),
        rules_engine=azure_native.network.SubResourceArgs(
            id="string",
        ),
        web_application_firewall_policy_link=azure_native.network.RoutingRuleUpdateParametersWebApplicationFirewallPolicyLinkArgs(
            id="string",
        ),
    )],
    tags={
        "string": "string",
    })

```
